### ISSUE TITLE
Multiple newsletter captchas on the same page do not work

### ISSUE THREAD

**Issue by iranimij** [iMi-digital/magento2-friendly-captcha#30]
I have two newsletter forms on my home page, but it seems that the module does not support two or multiple captcha on the same page. since we have this below code for newsletters!

     document.querySelector('.newsletter .subscribe').append(document.getElementById('<?= /* @noEscape */ $block->getWidgetId() ?>'));

You can see we just send one widget ID to the template and because of that, we can not show multiple captchas.

**Comment by iranimij** [comment]
Yes, But It seems that I do not have access to the repo.

**Comment by sprankhub** [comment]
@amenk I don't believe this is fixed. We use version `2.0.3` on a site and have two newsletter signup forms on the homepage, from which only one has a FriendlyCaptcha widget. I think this needs to be re-opened.

**Comment by amenk** [comment]
I think I closed this by accident.

One side note which partially might apply to this: It is not a good idea to have many FriendlyCaptcha widgets on the same page. We recently had a bug where we initialized ~ 60 such widgets on the same page for configurables with many variants and got chrome out of memory issues.

@sprankhub Two forms should not be a problem though .. but I am not sure if it leads to a nice layout with two newsletter signup forms on the same page? Did you consider something like one form where you can choose the newsletter to subscribe via ratio buttons, checkboxes or a dropdown list?

**Comment by sprankhub** [comment]
I understand that 60 might lead to issues, but I also think two shouldn't be an issue :D 

Not sure if I understand your idea. Currently, the design is just that way. Somewhere within the CMS page, you can sign up or in the footer :man_shrugging:
